### PR TITLE
Map symbols to operations and expressions

### PR DESCRIPTION
**Pull request by tealeg**
This PR is in support of #17 (I've started pulling bits out of the parser branch so you can review them in small chunks). 

Add a declarative mechanism for defining the mapping between symbols in our symbolic expression language and the operator implementations in rule package.  Crucially we do this without exposing the operator implementations, thus maintaining the privacy of that package (as per the previous release).

### CODE REVIEW

**Comment by asdine** [comment]
@tealeg ⚠️ The build is failing

File: rule/contract.go [hunk 1]
```diff
@@ -70,3 +72,22 @@ type Contract struct {
 type TypedExpression interface {
 	Contract() Contract
 }
+
+// GetOperatorExpr returns an Expr that matches the provided operator
+// name. If no matching expression exists, and error will be returned.
```
Comment (asdine) [inline rule/contract.go]:
```suggestion
// name. If no matching expression exists, an error will be returned.
```

File: rule/contract.go [hunk 2]
```diff
@@ -70,3 +72,22 @@ type Contract struct {
...
+		return &exprNot{}, nil
+	case "and":
+		return &exprAnd{}, nil
+	case "or":
+		return &exprOr{}, nil
+	case "in":
+		return &exprIn{}, nil
+
+	}
+	return nil, fmt.Errorf("No operator Expression called %q exists", name)
```
Comment (asdine) [inline rule/contract.go]: 🌷 🌷 Error strings should not be capitalized

Comment (tealeg) [reply rule/contract.go]: @YassineE agreed - and picking me up on not doing the right thing is a big part of what review is for.  I don't need flowers ever time you notice that I'm not good at these small things ;-)

**Pull request merged**